internal/pkg/manifest/validate.go

```diff
@@ -588,6 +589,24 @@ func (s StaticSiteConfig) validate() error {
 }
 
 func (f FileUpload) validate() error {
+	if err := f.validateSource(f.Source); err != nil {
+		return err
+	}
+	return nil
+}
+
+// validateSource returns nil if Source is configured correctly.
+func (f FileUpload) validateSource(source string) error {
+	if source == "" {
+		return &errFieldMustBeSpecified{
+			missingField: "source",
+		}
+	}
+	fs := afero.NewOsFs()
+	_, err := fs.Stat(f.Source)
+	if err != nil {
+		return fmt.Errorf("source '%s' must be a valid path", f.Source)
+	}
```

Comment:
Can we guarantee that the `err` can always be attributed to the path being invalid? 

In addition, small nit:
```suggestion
	if err != nil {
		return fmt.Errorf("source '%s' must be a valid path: %w", f.Source, err)
	}
```

Reply:
> Stat returns the FileInfo structure describing file. If there is an error, it will be of type *PathError.

Oh I see from https://github.com/spf13/afero/blob/d92c300a1b6eaa1690b7aecab2ef7fd0d64a94e1/os.go#L86 and https://pkg.go.dev/os#File.Stat that the err will always be a `*PathError` for now 👍🏼